cost/google/idle_compute_instances/google_idle_compute_instances.pt

```diff
@@ -209,77 +216,102 @@ script "js_calculated_utilization", type: "javascript" do
-        var memory_average = "101"
-        var memory_minimum = "101"
-      if ( memory_record === null || memory_record === undefined ) {
-        // No memory records, continue
-      } else {
-        if ( memory_record.mem_points === null || memory_record.mem_points === undefined ) {
-          // No memory records, continue
-        } else {
-          var memory_maximum = parseFloat( memory_record.mem_points.reduce( function (x, y) { return Math.max(x, y) } )).toFixed(2)
-          var memory_sum = _.reduce(memory_record.mem_points, function(memo, num){ return memo + num; }, 0);
-          var memory_average = parseFloat(memory_sum/memory_record.mem_points.length).toFixed(2)
-          var memory_minimum = parseFloat( memory_record.mem_points.reduce( function (x, y) { return Math.min(x, y) } )).toFixed(2)
-        }
-      }
-
-      results.push(
-        {
-          zone: ds_compute_utilization[i].zone,
-          projectId: ds_compute_utilization[i].projectId,
-          instance_id: instance_id,
-          cpu_average: cpu_average,
-          cpu_maximum: cpu_maximum,
-          cpu_minimum: cpu_minimum,
-          mem_average: memory_average,
-          mem_maximum: memory_maximum,
-          mem_minimum: memory_minimum
-        }
-      )
+    if (instance_id === null || instance_id === undefined) {
+      continue
+    }
+    var memory_record = _.find(ds_memory_utilization, function(record) { return record.instance_id == instance_id; })
+    var mem_average = "101"
+    var mem_maximum = "101"
+    var mem_minimum = "101"
+    if (memory_record !== null && memory_record !== undefined) {
+      mem_average = memory_record.mem_average
+      mem_maximum = memory_record.mem_maximum
+      mem_minimum = memory_record.mem_minimum
     }
+    results.push({
+      zone: ds_compute_utilization[i].zone,
+      projectId: ds_compute_utilization[i].projectId,
+      instance_id: instance_id,
+      cpu_average: ds_compute_utilization[i].cpu_average,
+      cpu_maximum: ds_compute_utilization[i].cpu_maximum,
+      cpu_minimum: ds_compute_utilization[i].cpu_minimum,
+      mem_average: mem_average,
+      mem_maximum: mem_maximum,
+      mem_minimum: mem_minimum
+    })
   }
 EOS
 end
 
+script "js_grouped_project_ids", type: "javascript" do
+  result "grouped_project_ids"
+  parameters "ds_calculated_utilization"
+  code <<-EOS
+  var grouped_project_ids = _.keys(_.groupBy(ds_calculated_utilization, "projectId"))
+  EOS
+end
+
+script "js_collected_instances", type: "javascript" do
+  result "collected_instances"
+  parameters "ds_instances"
+  code <<-EOS
+  var collected_instances = []
+  for (var i = 0; i < ds_instances.length; i++) {
+    for (var j = 0; j < ds_instances[i].instances.length; j++) {
+      collected_instances.push(ds_instances[i].instances[j])
+    }
+  }
+  EOS
+end
+
 script "js_add_instance_data", type: "javascript" do
-  result "request"
-  parameters "project","zone", "instance_id"
+  result "instances_with_data"
```

Comment:
parameters line should be above result line:
https://flexera.atlassian.net/wiki/spaces/FP/pages/2184642622/Policy+Best+Practices#Scripts

Reply:
@XOmniverse thank you for this, I addressed this change